### PR TITLE
Fix MariaDB support in `FactoryTest`

### PR DESCRIPTION
**Pull request by bwaidelich**
Apparently MariaDB does not contain the
```
 (using password: YES)
```
substring in the authentication error message.
By making that substring optional in the corresponding regex check, the tests are green for me (11.1.2-MariaDB @ Macbook Pro M1)

### CODE REVIEW

**Review by SimonFrings — COMMENTED** [review]
@bwaidelich Thanks for opening this pull request :+1:

This seems like a fairly simple change and brings up the bigger question, how are we making sure this works as expected across different relational database management systems? For now I can only take your word for it, as we don't have proof that this test failed with MariaDB, so I think we need to improve our test suite to cover these cases.

I'm not sure if this is something to discuss for this pull request, as answering this question takes up much more time and I don't want to leave this laying around for too long.

The test suite is currently green and was green before, so we could argue that this doesn't break anything we're currently testing against and merge it as is. I personally can't really "approve" this PR as it's hard to proof what's currently getting fixed, so maybe postpone until we have tests in place?

Interested in your and @WyriHaximus's and @clue's opinion on this.

**Comment by bwaidelich** [comment]
>  how are we making sure this works as expected across different relational database management systems?

I was wondering the same.
You could maybe add MariaDB to the CI matrix as it's a fairly common MySQL replacement

**Comment by WyriHaximus** [comment]
> Interested in your and @WyriHaximus's and @clue's opinion on this.

TL;DR: here is my opinion on this: https://github.com/friends-of-reactphp/mysql/pull/196

> The test suite is currently green and was green before, so we could argue that this doesn't break anything we're currently testing against and merge it as is. I personally can't really "approve" this PR as it's hard to proof what's currently getting fixed, so maybe postpone until we have tests in place?

We should increase our compatibility matrix. I've just put up https://github.com/friends-of-reactphp/mysql/pull/196 as to how we will achieve that. It's rough on the edges but it is the way forward IMHO. The question is should `MariaDB` be in that PR or a follow up. Funny enough with `MariaDB` it's still green. However, only for the previous major version of both, doing both previous and current mayor version and tests start failing: https://github.com/friends-of-reactphp/mysql/actions/runs/8381598989

**Comment by bwaidelich** [comment]
Closing to remove this one from my list of pending PRs
